dronekit/__init__.py

```diff
@@ -3012,7 +3012,7 @@ def connect(ip,
 
         @vehicle.on_message('STATUSTEXT')
         def listener(self, name, m):
-            status_printer(re.sub(r'(^|\n)', '>>> ', m.text.rstrip()))
+            status_printer(re.sub(r'(^|\n)', '>>> ', m.text.decode("utf-8").rstrip()))
```

Comment:
There's no guarantee that m.text is valid unicode data.  In fact, I'm investigating a case ATM where it is not (but should be)!

I'm concerned that after this patch we might end up with failed-to-decode exceptions of some kind here.

I think we'll have to accept that we can't use a regex here and find some other mechanism to do the sub - perhaps even just assume all statustext lines are single-line-only (which you kind of have to if you don't know what the text encoding is)!